### DISCUSSION TITLE
How would you make other hooks work with InstantSearchSSRProvider and Next.js?

### DISCUSSION THREAD

**Discussion by landermedia** [algolia/instantsearch#5689] [Q&A]
I've been really scratching my head over this as I couldn't make things other things such as other React context's. Here's [a demo of InstantSearchSSRProvider with next-auth](https://codesandbox.io/p/sandbox/fervent-banzai-z59qnn?file=%2Fpages%2Findex.tsx%3A77%2C10-77%2C27) but I get an error such below

<img width="650" alt="image" src="https://github.com/algolia/instantsearch/assets/83575285/0289289e-00ae-420c-bdef-90b24dcc7752">

I also did tried to use a separate `ChildComponent` class as seen in `index2.tsx` but to no avail. Forgive my inexperience but you could point me in the right direction here if this isn't a bug? Thanks!

PS: I've got more other hooks that seems to not recognize the context anymore in the real project where I encountered this issue. If I remove the `getServerSideProps`, everything works well.

**Comment by Haroenv** [comment]
The component you're passing to `getServerState` needs to be standalone. If your application can't rely on making that context optional, by using a fallback for useSession for example, what you can do is take only the InstantSearch widgets that will be mounted, create a new component for it and render that:

```js
function AppForSSR() {
  return <InstantSearchSSRProvider ...>
    <SearchBox />
    <RefinementList />
    ...


getServerState(<AppForSSR />)
```

**Reply by landermedia** [reply]
Thank you @Haroenv for your prompt response. I'll get back to you on this as soon as I can try your suggestion.
